### PR TITLE
refactor: remove `influxdb_iox import` and related code

### PR DESCRIPTION
**Pull request by alamb**
# Rationale:
The `influxdb_iox import` code was part of a project to prepare for tsm migration. However, our plans have changed and this code is no longer needed. 

@domodwyer  also says in https://github.com/influxdata/influxdb_iox/pull/7799#pullrequestreview-1428764314

> I think we can bin this - it's completely unused, and if we need to reference it we can go digging in the history!

> I've noticed it's been a bit of a burden on @carols10cents as she refactors some of the types this code uses for the partitioning changes, despite this code being unused.

Internal slack reference: https://influxdata.slack.com/archives/C04QYLRFQE5/p1684245013760149

(also I am trying to catch up to @carols10cents deleting stats)

# Changes
1. remove `influxdb_iox import` and related code

### CODE REVIEW

File: influxdb_iox/src/main.rs [hunk 1]
```diff
@@ -373,13 +369,6 @@ fn main() -> Result<(), std::io::Error> {
                     std::process::exit(ReturnCode::Failure as _)
                 }
             }
-            Some(Command::Import(config)) => {
```
Comment (alamb) [inline influxdb_iox/src/main.rs:376]: The fact that there is no end to end test for this command is further evidence that we didn't quite finish it

**Review by domodwyer — APPROVED** [review]

**Comment by crepererum** [comment]
Does this mean we can also wipe https://github.com/influxdata/influxdb_iox/tree/main/influxdb_tsm ?

**Comment by domodwyer** [comment]
Can for me - same logic applies - it lives in git :+1:

**Comment by alamb** [comment]
> Does this mean we can also wipe https://github.com/influxdata/influxdb_iox/tree/main/influxdb_tsm ?

🤔  I started hacking on a way to convert TSM to line protocol in https://github.com/influxdata/influxdb_iox/pull/7418 (which I think would be quite useful for testing / small scale deployments) that uses that code,  but I haven't gotten very far

**Comment by crepererum** [comment]
> > Does this mean we can also wipe https://github.com/influxdata/influxdb_iox/tree/main/influxdb_tsm ?
> 
> thinking I started hacking on a way to convert TSM to line protocol in #7418 (which I think would be quite useful for testing / small scale deployments) that uses that code, but I haven't gotten very far

I'm a bit confused by the referenced PR because all the methods are called "parquet to line protocol", but the PR title suggests something else and some code mentions TSM. So what is converted to what there? :thinking:

**Review by carols10cents — APPROVED** [review]

**Comment by alamb** [comment]
Hearing no dissent for this plan, merging!

**Pull request merged**